executor/src/main/java/com/redhat/service/bridge/executor/Executor.java

```diff
@@ -15,14 +20,30 @@
 
     private final ProcessorDTO processor;
 
-    public Executor(ProcessorDTO processor) {
+    private final Set<String> templates;
+
+    private final FilterEvaluator filterEvaluator;
+
+    private final TemplateFactory templateFactory;
```

Comment:
I don't like this `template` intermediate step, I would hide the `TemplateFactory` inside the `FilterEvaluator` and accept `Set<BaseFilter>` directly.
If you want to "parse" the filter only once you can rename `FilterEvaluator` as `FilterEvaluatorBuilder` and adds a `FilterEvaluatorInstance` concept that is a specific instance tied to a set of filters.
Current decoupling implies that `TemplateFactory` and `FilterEvaluator` are related (both needs to be the FEEL version) but this is not enforced in the code.
Wdyt?

Reply:
Fair point. I've refactored the code with the following logic: 

```java
public interface FilterEvaluatorFactory {
    FilterEvaluator build(Set<BaseFilter> filters);
}
```
and 
```java
public interface FilterEvaluator {
    boolean evaluateFilters(Map<String, Object> data);
}
```

And we provide a FEEL implementation `FilterEvaluatorFactoryFEEL` and `FilterEvaluatorFEEL`. The `Executor` is simply built with ` new Executor(processorDTO, filterEvaluatorFactory.build(processorDTO.getFilters()))` and it's completely trasparent for the consumer 

```java
        if (filterEvaluator.evaluateFilters(CloudEventUtils.getMapper().convertValue(cloudEvent, Map.class))) {
            LOG.info("[executor] Filters of processor '{}' matched for event with id '{}'", processor.getId(), cloudEvent.getId());
            // TODO - CALL ACTIONS;
            // TODO - consider if the CloudEvent needs cleaning up from our extensions before it is handled by Actions
            return;
        }
```